src/key/rsa.js

```diff
@@ -0,0 +1,32 @@
+const { generateKeyPair } = require('crypto');
+const pemtools = require('pemtools');
+
+export const generateRSAKeyPair = (length = 4096) => {
+	return new Promise((resolve, reject) => {
+		generateKeyPair(
+			'rsa',
+			{
+				modulusLength: length,
+				publicKeyEncoding: {
+					type: 'spki',
+					format: 'pem'
+				},
+				privateKeyEncoding: {
+					type: 'pkcs1',
+					format: 'pem'
+				}
+			},
+			(err, publicKey, privateKey) => {
+				if (err) {
+					return reject(err);
+				}
+				return resolve({ privateKey, publicKey });
+			}
+		);
+	});
+};
+
+export const getPEMBits = privateKey => {
+	const pem = pemtools(privateKey);
```

Comment:
Although I was the one that asked for a secret size check, I'm not sure it is worth it to bring in a third party dependency just to check this. More so if we are providing the key generation tooling.

Also, it is third party code reading private keys. Have considered alternatives? Maybe checking the minimum pem size directly?

Reply:
Leaving it for now, adding todo to reconsider